### PR TITLE
TravisCI runs integration tests

### PR DESCRIPTION
**Pull request by ca-johnson**
* Run make integration tests
* Install integration test dependencies in install_linux.sh
* cd instead of pushd to fix `pushd not found` on travis agent
* Set P4Repo(root=) to avoid stray `p4config` file being written into cwd (which can affect integration tests, since it picks up nearby p4configs)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 local_run: vendorize p4d
 	rm -f -r p4_workspace
 	mkdir local-pipeline -p
-	pushd local-pipeline && bk local run ../.buildkite/local-pipeline.yml --meta-data "buildkite-perforce-revision=@6"
+	cd local-pipeline && bk local run ../.buildkite/local-pipeline.yml --meta-data "buildkite-perforce-revision=@6"
```
Comment (stefan-improbable) [inline Makefile:4]: nit: the `bk local run` could probably be on it's own line, not sure why there is `&&`

File: ci/install_linux.sh [hunk 2]
```diff
@@ -2,13 +2,23 @@
 set -ex
 sudo apt-get install python-dev build-essential libssl-dev
 
+# Install p4d (perforce server)
```
Comment (stefan-improbable) [inline ci/install_linux.sh:5]: I recommend using `set -euxo pipefail`, it's a bit better.

File: ci/install_linux.sh [hunk 3]
```diff
@@ -2,13 +2,23 @@
...
 wget http://www.perforce.com/downloads/perforce/r18.2/bin.linux26x86_64/p4api.tgz
-# Detect concrete version number inside p4api.tgz
+# Detect concrete version number inside p4api.tgz, changes as new versions are published
 P4API_VERSION=$(tar -tf p4api.tgz | head -1)
 tar xzf p4api.tgz --directory /tmp/
 wget https://files.pythonhosted.org/packages/36/5a/0a1b192cdecd31cb8bc0d0ba39c73ffd84ce823053d0004823a1fdbe1440/p4python-2018.2.1743033.tar.gz
 tar xfz p4python-2018.2.1743033.tar.gz --directory /tmp/
-pushd /tmp/p4python-2018.2.1743033/ && python setup.py install --apidir /tmp/${P4API_VERSION} && popd
\ No newline at end of file
+pushd /tmp/p4python-2018.2.1743033/ && python setup.py install --apidir /tmp/${P4API_VERSION} && popd
```
Comment (stefan-improbable) [inline ci/install_linux.sh]: no need for &&, you can put these on separate lines and it will be more readable

**Review by stefan-improbable — APPROVED** [review]
lgtm % nits

File: Makefile (hunk 1 above)
Comment (ca-johnson) [reply Makefile:4]: each line in makefile runs in its own shell 🤷‍♂

File: ci/install_linux.sh (hunk 3 above)
Comment (ca-johnson) [reply ci/install_linux.sh]: updated throughout

File: ci/install_linux.sh (hunk 2 above)
Comment (ca-johnson) [reply ci/install_linux.sh:5]: done, thx

**Pull request merged**